trussworks/react-uswds
# ci: Migrate to Github Actions

# Summary

The results of an investigation into #1688 thus far. Some hiccups were encountered that will require further digging

## Related Issues or PRs

Closes: #1688

## How To Test

- Verify that all CI steps run properly. 
- Make sure that the considerations in the Remaining TODOs and Blockers are addressed. 
  - Note that things like dependabot PRs are harder to replicate and test
- Verify that duplication of work by the CI is kept to a minimum, and that anything migrated to Actions is no longer being performed in Circle

### Remaining TODOs and blockers:

- [x] **TODO**: The process of automatically merging approved dependabot PRs will likely break until further changes are made to secret management and permissions.
  - We can ensure that the CI runs for dependabot PRs by creating a two-workflow system ([suggested in this comment](https://github.com/dependabot/dependabot-core/issues/3253#issuecomment-797125425))
  - We would then need to verify that kodiak could still auto-merge properly.
  - See https://github.com/dependabot/dependabot-core/issues/3253 for potential workarounds (and hopefully a more robust solution in the near future as indicated in [this comment](https://github.com/dependabot/dependabot-core/issues/3253#issuecomment-940182533))
  - **Note**: This will be evaluated when we see dependabot PRs breaking
- [x] **TODO** (Maybe blocker?: Ephemeral deployments work with CircleCI specifically because Circle serves the uploaded artifacts, where Github does not. These can either remain in Circle (as this PR does so far) OR some additional infrastructure setup/management would be required to mimic the functionality.
  - We could use the [storybook-deployer](https://github.com/storybookjs/storybook-deployer) that we already use for deployments to github pages on tag creation (see the deploy-storybook.yml workflow). This allows you to deploy to S3. If we could configure prefixes involving the commit SHA for uniqueness, we could build URLs to static storybook instances.
  - **Note**: Leaving these to run in Circle CI for now
- [x] **TODO**: For Happo to work properly in this PR, the Happo secrets must be added to Github secrets (Currently there is a false positive)
  - [x] These are available under `Account` in Happo
  - [ ] MilMove uses Happo, double check with them what their configuration is
